app/components/autocomplete/at_mention/at_mention.tsx

```diff
@@ -289,15 +304,25 @@ const AtMention = ({
             const filteredUsers = filterResults(fallbackUsers, term);
             setFilteredLocalUsers(filteredUsers.length ? filteredUsers : emptyUserlList);
         } else if (receivedUsers) {
-            if (hasTrailingSpaces(term)) {
-                const filteredReceivedUsers = filterResults(receivedUsers.users, term);
-                const filteredReceivedOutOfChannelUsers = filterResults(receivedUsers.out_of_channel || [], term);
+            const memberIds = receivedUsers.users.map((e) => e.id);
+            const sortedMembers: Array<UserProfile | UserModel> = await getUsersFromDMSorted(sUrl, memberIds);
+            const sortedMembersId = sortedMembers.map((e) => e.id);
+            const membersNoDm = receivedUsers.users.filter((u) => !sortedMembersId.includes(u.id));
```

Comment:
you could should make `sortedMembersId` a Set so that the filter is more performant